File: mobile_client/lib/model/Utente.dart
class Utente {
  int id;
  String email;
  String nome;
  String cognome;
  String token;
  DateTime dataCreazione;
  int idRuolo;

  Utente({
    required this.id,
    required this.email,
    required this.nome,
    required this.cognome,
    required this.token,
    required this.dataCreazione,
    required this.idRuolo,
  });

  // Convert a Utente into a JSON-formatted Map.
  Map<String, dynamic> toJson() => {
        'id': id,
        'email': email,
        'nome': nome,
        'cognome': cognome,
        'token': token,
        'data_creazione': dataCreazione.toIso8601String(),
        'id_ruolo': idRuolo,
      };

  // Create a Utente from a JSON-formatted Map.
  factory Utente.fromJson(Map<String, dynamic> json) => Utente(
        id: json['id'],
        email: json['email'],
        nome: json['nome'],
        cognome: json['cognome'],
        token: json['token'],
        dataCreazione: DateTime.parse(json['data_creazione']),
        idRuolo: json['id_ruolo'],
      );
}

File: helpdesk_API/model/Utente.php
<?php

namespace model;

class Utente
{
    public $id;
    public $email;
    public $nome;
    public $cognome;
    public $data_creazione;
    public $id_ruolo;

    public function __construct($id, $email, $nome,$cognome,$data_creazione,$id_ruolo)
    {
        $this->id = $id;
        $this->email = $email;
        $this->nome = $nome;
        $this->cognome = $cognome;
        $this->data_creazione = $data_creazione;
        $this->id_ruolo = $id_ruolo;
    }


}